tests/test_numpy_array.py

```diff
@@ -377,3 +377,18 @@ def test_array_unchecked_dyn_dims(msg):
 
     assert proxy_auxiliaries2_dyn(z1) == [11, 11, True, 2, 8, 2, 2, 4, 32]
     assert proxy_auxiliaries2_dyn(z1) == array_auxiliaries2(z1)
+
+def test_array_resize(msg):
```

Comment:
You should also test some failure conditions here, i.e. that you get the expected exception and exception message if attempting to resize a referenced array.